src/queries.js

```diff
@@ -136,7 +137,6 @@ function queryAllByTitle(
 function queryByTitle(...args) {
   return firstResultOrNull(queryAllByTitle, ...args)
```

Comment:
So the query APIs would still return the first one even if there are multiple?

Reply:
Correct. Hmmm... Should we throw for this as well? Probably huh...

Reply:
Not sure... querySelector vs querySelectorAll would just return the first item

Reply:
I could see it throwing as well. Maybe to clarify it all we could make a table:

|            | One Match | Multiple Matches | No Match | Async? |
| ---------- | --------- | ---------------- | -------- | ------ |
| getBy      | Return    | Throw            | Throw    | No     |
| findBy     | Return    | Throw            | Throw    | Yes     |
| queryBy    | Return    | Throw            | Null    | No     |
| getAllBy   | Array     | Array            | Throw    | No     |
| findAllBy  | Return    | Throw            | Throw    | Yes     |
| queryAllBy | Array     | Array            | Array    | No     |

Reply:
In this model does it make sense to have `query` and `queryAll`? Should there just be:

`query`
`get`
`getAll`
`find`
`findAll`?

To me that seems pretty consistent with like, the type of behavior you'd expect from a REST API for example. like `/api/users` is going to return a list of users even if there's only 1 (or 0). But if i say `/api/users/1` or `/api/users/1,2` i'm asking for specific things and I'll get a 404 if they're not found.

Reply:
The current API is based on querySelector/querySelectorAll

https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector

> The Document method querySelector() returns the first Element within the document that matches the specified selector, or group of selectors. If no matches are found, null is returned.

https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll

> A non-live NodeList containing one Element object for each element that matches at least one of the specified selectors or an empty NodeList in case of no matches.

Reply:
👌 makes sense now, thanks for that explanation. Does querySelector throw if more than 1 is found?

Reply:
> Does querySelector throw if more than 1 is found?

No, it returns the first element (see above).

Reply:
🙄 duh it was the first sentence, nvm i can't read apparently.

Reply:
I feel like I'd like to see query continue to never throw. I think it's nice to have a set of searches that you know aren't going to throw in any circumstance, but then again it might catch some bugs people don't know about currently... I dunno 🤔

Do you see any chance that throwing might make it more confusing for knowing when to use query vs. get since they both have the potential to throw in this model?

Reply:
If you think there could be more than one then use queryAll*

Reply:
The primary use case for `queryBy` is for when you think/expect something not to exist and you want to check it returns null. The only I can think of it being disruptive to change it would be if you were doing something with `.not`, like `wait(() => expect(queryByText('something')).not.toBeNull()`.

Reply:
sounds good to me, i can dig it. we can make this happen in native too 👍